src/node.ts

```diff
@@ -97,10 +102,21 @@ export class EthereumNode extends constructs.Construct {
     // will be populated with defaults when passed to the node constructor
     this.network = props.network ?? Network.MAINNET;
     this.instanceType = props.instanceType ?? InstanceType.BURSTABLE3_LARGE;
-    this.availabilityZone = props.availabilityZone ?? `${region}a`;
 
-    utilities.validateRegion(region);
-    utilities.validateAvailabilityZone(region, this.availabilityZone);
+
+    if (tokenRegion) {
```

Comment:
Here's how I'd suggest simplifying. See how the code is almost completely self-documenting now, no need for detailed comments.

```
// If no availability zone is provided, just use the first one.
this.region = props.region ?? region;
this.availabilityZone = props.availabilityZone ?? availabilityZone[0];

if (regionInEnvironment) {
    utilities.validateRegion(region);
    utilities.validateAvailabilityZone(region, this.availabilityZone);
}
```

Reply:
I made a small change here to accommodate an edge case: 

```
// If no availability zone is provided, use the first in the region.
if (regionInEnvironment) {
    utilities.validateRegion(region);
    this.availabilityZone = props.availabilityZone ?? `${region}a`;
    utilities.validateAvailabilityZone(region, this.availabilityZone);
} else {
    this.availabilityZone = props.availabilityZone ?? availabilityZones[0];
}
```

If `region` is specified in the environment variables, we want to assign the real availability zone value and then validate it, and if no availability zone is specified, assign the first in the region. If left unspecified, you wouldn't be able to assign the first token in the availability zone array AND validate at the `cdk synth` simultaneously.